src/EditorFeatures/Core/Implementation/Classification/SemanticClassificationUtilities.cs

```diff
@@ -34,6 +34,12 @@ internal static class SemanticClassificationUtilities
             IClassificationService classificationService,
             ClassificationTypeMap typeMap)
         {
+            // Let LSP handle semantic classification when running in the LSP editor.
+            if (spanToTag.SnapshotSpan.Snapshot.TextBuffer.IsInLspEditorContext())
```

Comment:
Could you explain why this wasn't needed before in a normal LiveShare scenario? Aka which piece did the "IsLiveShare" logic to ensure that classification didn't apply there

Reply:
Files go into a separate workspace in the client side + theres no real semantic model on the client.  When we use this locally however, neither of those are true